grammars/coffeescript.cson

```diff
@@ -12,14 +12,66 @@
 'foldingStopMarker': '^\\s*$|^\\s*[}\\]]\\s*$'
 'patterns': [
   {
+    'comment': 'functions without parameters: `->`'
     'captures':
       '1':
-        'name': 'variable.parameter.function.coffee'
+        'name': 'punctuation.definition.parameters.begin.coffee'
+      '2':
+        'name': 'punctuation.definition.parameters.end.coffee'
+      '3':
+        'name': 'storage.type.function.coffee'
+    'match': '(?x)
+      (?:
+        (\\()
+        \\s*
+        (\\))
+      )?
+      \\s*
+      ([=-]>)
+    '
+    'name': 'meta.inline.function.without-parameters.coffee'
+  }
+  {
+    'comment': 'functions with parameters: `(param1, @param2) ->`'
+    'begin': '(?x)
+    (\\()
+    (?=
+      (?:
+        @?[a-zA-Z\\$_]\\w*|,|\\s
+      )*
+      \\)
+      \\s*
+      [=-]>
+    )
```

Comment:
From TextMate Manual:
> Note that the regular expressions are matched against only a single line of the document at a time. That means it is not possible to use a pattern that matches multiple lines.

I don't know is it possible in Atom.

Lookahead `?=` can't find `->` in the next line. That's why multiline parameters don't work.